models/Element/AbstractElement.php

```diff
@@ -172,7 +172,9 @@ public function isLocked()
         }
 
         // check for inherited
-        return $this->getDao()->isLocked();
+        /** @var Model\Document\Dao|Model\Asset|Dao|Model\DataObject\AbstractObject\Dao $dao */
+        $dao = $this->getDao();
```

Comment:
I think we should use here a method comment on class. Like https://github.com/pimcore/pimcore/pull/5765/commits/dd826788a0bcfbdeab6c759ae0f0b67138ea2970